### PR TITLE
Add support for the ARM64 Linux platform

### LINKED ISSUE #1367: Buid Kani on arm linux
We've got Kani building on M1, now let's get it building on arm linux.

Out of scope is CI/install, just like M1.

### LINKED ISSUE #2718: Concrete playback prints characters for interpreted values in ARM64 Linux
I tried this code:

Applies to all the tests listed in [this comment](https://github.com/model-checking/kani/issues/1367#issuecomment-1692241746). For example, `tests/ui/concrete-playback/u8/main.rs`:
```rust
#[kani::proof]
pub fn harness() {
    let u8_1: u8 = kani::any();
    let u8_2: u8 = kani::any();
    let u8_3: u8 = kani::any();
    assert!(!(u8_1 == u8::MIN && u8_2 == 101 && u8_3 == u8::MAX));
}

```

using the following command line invocation: `kani --enable-unstable --concrete-playback=print`
with Kani version: 0.34

I expected to see this happen: Concrete playback generates the test
```rs
#[test]
fn kani_concrete_playback_harness_6199617291819462955() {
    let concrete_vals: Vec<Vec<u8>> = vec![
        // 0
        vec![0],
        // 101
        vec![101],
        // 255
        vec![255]
    ];
    kani::concrete_playback_run(concrete_vals, harness);
}
```

Instead, this happened: In ARM64 Linux, concrete playback generates the test
```rs
#[test]
fn kani_concrete_playback_harness_6199617291819462955() {
    let concrete_vals: Vec<Vec<u8>> = vec![
        // 0
        vec![0],
        // 'e'
        vec![101],
        // 255
        vec![255],
    ];
    kani::concrete_playback_run(concrete_vals, harness);
}
```
Note that the interpreted value for the byte with value 101 is `'e'`. @tautschnig notes in [this comment](https://github.com/model-checking/kani/issues/1367#issuecomment-1692743287) that this comes from CBMC.

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

This PR adds support for the `aarch64-unknown-linux-gnu` platform (ARM64 Linux). It does not include installation nor CI workflows, only manually-tested support.

The main complication came from the concrete playback feature (#2718), which returns characters instead of numbers for interpreted values in some cases. This PR includes a fix to convert them into their ASCII values. 

### Resolved issues:

Resolves #1367
Resolves #2718 

### Testing:

* How is this change tested? Manually, with existing tests on an ARM64 machine.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by jaisnan — APPROVED** [review]
LGTM! But I wonder if we should include documentation changes to the same PR?

File: kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs [hunk 1]
```diff
@@ -690,12 +697,15 @@ fn new_machine_model(sess: &Session) -> MachineModel {
...
+            let long_double_width = match os.as_ref() {
+                "linux" => 128,
+                _ => 64,
+            };
             let long_int_width = 64;
             let long_long_int_width = 64;
             let short_int_width = 16;
             let single_width = 32;
-            let wchar_t_is_unsigned = false;
+            let wchar_t_is_unsigned = matches!(os.as_ref(), "linux");
```
Comment (zhassan-aws) [inline kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:708]: Any reference we can cite?

**Review by zhassan-aws — APPROVED** [review]

File: kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs (hunk 1 above)
Comment (adpaco-aws) [reply kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:708]: It's quite hard find (and link) this information in ARM manuals, which is why I'd suggest the first two ([here](https://developer.arm.com/documentation/dui0491/i/Compiler-Command-line-Options/--signed-chars----unsigned-chars) and [here](https://www.arm.linux.org.uk/docs/faqs/signedchar.php)) for Linux and the last one ([here](https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms)) for Apple.
```suggestion
            // https://developer.arm.com/documentation/dui0491/i/Compiler-Command-line-Options/--signed-chars----unsigned-chars
            // https://www.arm.linux.org.uk/docs/faqs/signedchar.php
            // https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms
            let wchar_t_is_unsigned = matches!(os.as_ref(), "linux");
```
What do you think?

Comment (zhassan-aws) [reply kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs:708]: Looks good. Thanks for digging up the references!

**Pull request merged**